src/backend/model/users.js

```diff
@@ -124,9 +161,89 @@ async function getRating(StudentID) {
 
     AverageRating = await db.query(avgRatingQuery);
 
+    for (let i = 0; i < Ratings.rows.length; i++) {
+        Ratings.rows[i].StudentID = parseInt(Ratings.rows[i].studentid);
+        Ratings.rows[i].StarsCount = parseInt(Ratings.rows[i].starscount);
+        Ratings.rows[i].Review = Ratings.rows[i].review;
+        delete Ratings.rows[i].studentid;
+        delete Ratings.rows[i].starscount;
+        delete Ratings.rows[i].review;
+    }
+
 
     return {
         Ratings: Ratings.rows,
         AverageRating: AverageRating.rows[0].avgrating
     }
+}
+
+async function getPurchaseRequests(userID) {
+
+    const query = {
+        text: `SELECT pr.UsedBookID, pr.RequestTimestamp, b.Title As BookName, 
+        CASE WHEN p.BuyerID IS NULL THEN 'Pending'
+        WHEN p.BuyerID = $1 THEN 'Purchased'
+        ELSE 'Denied'
+        END AS Status
+        FROM purchaserequest AS pr
+        INNER JOIN usedbook AS ub ON pr.UsedBookID = ub.UsedBookID
+        LEFT JOIN book AS b ON ub.BookID = b.ISBN
+        LEFT JOIN purchase AS p ON pr.UsedBookID = p.UsedBookID
+        WHERE pr.BuyerID = $1`,
+        values: [ userID ],
+    };
+
+    return db.query(query)
+        .then((result) => {
+
+            for (let i = 0; i < result.rows.length; i++) {
+                result.rows[i].UsedBookID = parseInt(result.rows[i].usedbookid);
+                result.rows[i].RequestTimestamp = result.rows[i].requesttimestamp;
+                result.rows[i].BookName = result.rows[i].bookname;
+                result.rows[i].Status = result.rows[i].status;
+                delete result.rows[i].usedbookid;
+                delete result.rows[i].requesttimestamp;
+                delete result.rows[i].bookname;
+                delete result.rows[i].status;
+
+                result.rows[i].RequestTimestamp = reformatTimestamp(result.rows[i].RequestTimestamp);
+
+            }
+
+            return result.rows;
+        })
+        .catch((err) => {
+            console.log(err);
+            return null;
+        });
+}
+
+async function getUsedBook(userID) {
+    
+        const query = {
+            text: `SELECT b.Title As BookName, ub.UsedBookID, ub.AskingPrice
+            FROM usedbook AS ub
+            INNER JOIN book AS b ON ub.BookID = b.ISBN
+            LEFT JOIN purchase AS p ON ub.UsedBookID = p.UsedBookID
+            WHERE ub.SellerID = $1 AND p.BuyerID IS NULL`,
+            values: [ userID ],
+        };
+    
+        return db.query(query)
+            .then((result) => {
+                for (let i = 0; i < result.rows.length; i++) {
+                    result.rows[i].AskingPrice = parseFloat(result.rows[i].askingprice);
```

Comment:
Why is `askingprice` a float and not an integer as in the schema?

Reply:
Thanks for reminding. I fixed it.